java/code/src/com/suse/manager/webui/services/SaltServerActionService.java

```diff
@@ -1593,10 +1596,34 @@ else if (profile instanceof KiwiProfile) {
     private Map<LocalCall<?>, List<MinionSummary>> scapXccdfEvalAction(
             List<MinionSummary> minionSummaries, ScapActionDetails scapActionDetails) {
         Map<LocalCall<?>, List<MinionSummary>> ret = new HashMap<>();
-        String parameters = "eval " +
-            scapActionDetails.getParametersContents() + " " + scapActionDetails.getPath();
+        Map<String, Object> pillar = new HashMap<>();
+        Matcher profile_matcher = Pattern.compile("--profile ((\\w|\\.|_|-)+)").matcher(scapActionDetails.getParametersContents());
+        Matcher rule_matcher = Pattern.compile("--rule ((\\w|\\.|_|-)+)").matcher(scapActionDetails.getParametersContents());
+        Matcher tailoring_file_matcher = Pattern.compile("--tailoring-file ((\\w|\\.|_|-)+)").matcher(scapActionDetails.getParametersContents());
+        Matcher tailoring_id_matcher = Pattern.compile("--tailoring-id ((\\w|\\.|_|-)+)").matcher(scapActionDetails.getParametersContents());
+
+        pillar.put("xccdffile", scapActionDetails.getPath());
+        if (profile_matcher.find()) {
+            pillar.put("profile", profile_matcher.group(1));
+        }
+        if (rule_matcher.find()) {
+            pillar.put("rule", rule_matcher.group(1));
+        }
+        if (tailoring_file_matcher.find()) {
+            pillar.put("tailoring_file", tailoring_file_matcher.group(1));
+        }
+        if (tailoring_id_matcher.find()) {
+            pillar.put("tailoring_id", tailoring_id_matcher.group(1));
+        }
+        if (scapActionDetails.getParametersContents().matches(".*--fetch-remote-resources.*")) {
+            pillar.put("fetch_remote_resources", true);
+        }
+        if (scapActionDetails.getParametersContents().matches(".*--remediate.*")) {
+            pillar.put("remediate", true);
+        }
```

Comment:
We need to provide the additional oval files somehow.
This is important to run the full SUSE STIG without to specify --fetch-remote-resource

Reply:
Ok, I'll add some more code to parse also the oval files :+1:

Reply:
thanks